Content.Server/Tools/Components/WelderComponent.cs

```diff
@@ -60,5 +60,12 @@ public class WelderComponent : SharedWelderComponent
         /// </remarks>
         [DataField("litMeleeDamageBonus")]
         public DamageSpecifier LitMeleeDamageBonus = new();
+
+        /// <summary>
+        ///     Whether the item is safe to refill while lit without exploding the tank.
+        /// </summary>
+        [DataField("tankSafe")]
+        public bool TankSafe = false; //I have no idea what I'm doing
```

Comment:
As a comment in the `DamageOnToolInteractComponent` says:
> // TODO: Remove this snowflake stuff, make damage per-tool quality perhaps?

Really the on-tool-interact damage should be based on tool qualities, and not checking for a welder component.
But without re-doing that & turning welder-component into lighter-component, this is probably a good enough measure for now.

Reply:
Then again, do lighters even need to be tank safe?

Even if they have a smaller flame, I wouldn't recommend re-fuelling them while they're lit. That's just a bad idea.

Reply:
My logic was balance rather than accuracy; welderbombing is easy enough as-is, I didn't want to remove the only restricting factor which is that welders are slightly rare.

I can remove the tanksafe stuff, but I don't think I'll be able to refactor how tank explosions work.